ui/src/redux/selectors/services.js

```diff
@@ -41,6 +41,20 @@ export const selectService = (state, domainName, serviceName) => {
         : {};
 };
 
+export const selectServiceTags = (state, domainName, serviceName) => {
+    return state.services.services &&
+        state.services.services[
+            getFullName(domainName, serviceDelimiter, serviceName)
+        ] &&
+        state.services.services[
+            getFullName(domainName, serviceDelimiter, serviceName)
+        ].tags
+        ? state.services.services[
+              getFullName(domainName, serviceDelimiter, serviceName)
+          ].tags
+        : {};
+};
```

Comment:
we should add a helper function 

```
function getSafe(fn, defaultVal) {
    try {
      return fn();
    } catch (e) {
      return defaultVal;
    }
  };
```
then we can just do 
```
return getSafe(state.services.services[
    getFullName(domainName, serviceDelimiter, serviceName)].tags, {});
```

Reply:
this is a great idea I think we can implement it on all the selectors.